gridcomps/History/Sampler/MAPL_TrajectoryMod_smod.F90

```diff
@@ -71,6 +72,15 @@
               label=trim(string) // 'var_name_lat:', _RC)
          call ESMF_ConfigGetAttribute(config, value=traj%var_name_time_full, default="", &
               label=trim(string) // 'var_name_time:', _RC)
+         call ESMF_ConfigGetAttribute(config, value=traj%use_NWP_1_file, default=0, &
+              label=trim(string)//'use_NWP_1_file:', _RC)
+         if (mapl_am_I_root()) then
+            if (traj%use_NWP_1_file == 1) then
+               write(6,105) 'WARNING: Traj sampler: use_NWP_1_file is ON'
+               write(6,105) 'WARNING: USER needs to check if observation file is fetched correctly'
+            end if
+            _ASSERT ( (traj%use_NWP_1_file==1).OR.(traj%use_NWP_1_file==0), 'use_NWP_1_file: wrong input')
+         end if
```

Comment:
1. The value of "1" for `use_NWP_1_file` is a "magic" number.  Please introduce named parameters for the allowed values of `use_NWP_1_file`.
2. The assertion can be made slightly clearer (and more extensible) as:
`_ASSERT( any(traj%use_NWP_1_file == [0, 1]), 'use_NWP_1_file: wrong input')`
(Again - the [0,1] should be replaced with parameter values.)

Reply:
Why not make it a LOGICAL instead of INTEGER then?

Reply:
You are right, thank you @tclune. I change this variable to logical and I found quite a few examples of using logical in the HISTORY.rc in MAPL, which should be and I should have searched for that. Now the code logic is as clear as it should be.